### PR TITLE
Add document describing changes done during the merge

### PR DESCRIPTION
**Pull request by frant-hartm**

### CODE REVIEW

**Review by gurbuzali — APPROVED** [review]
looks good, maybe we can add bootstrap/cli related changes too

File: docs/design/merge/01-merge-changes.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Merge Changes Summary
+
+The Hazelcast and Hazelcast Jet products were merged together to provide a
+seamless experience. Changes done during the merge of Hazelcast and Hazelcast
+Jet codebases are summarized in this document.
+
+Hazelcast Jet repository was merged into Hazelcast repository.
+
+The name Jet doesn't refer to standalone product anymore, but is kept when
+referencing the streaming engine part of the product - the Jet engine.
```
Comment (viliam-durina) [inline docs/design/merge/01-merge-changes.md:10]:
```suggestion
referring to the streaming engine part of the product - the Jet engine.
```

File: docs/design/merge/01-merge-changes.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  actually happened after the merge, but for users it is indistinguishable).`
+
+## Code Changes
+
+- JetNodeExtension was merged with DefaultNodeExtension
+- The Jet service is started when a HazelcastInstance is started
+- HazelcastInstance can return JetInstance via #getJetInstance (this will
+  change before the release into a `JetService getJet()` where `JetService`
+  provides a subset of `JetInstance` methods)
+- Jet datastructures created by com.hazelcast.jet.impl.JobRepository are
```
Comment (viliam-durina) [inline docs/design/merge/01-merge-changes.md:48]:
```suggestion
- Jet data structures created by com.hazelcast.jet.impl.JobRepository are
```

File: docs/design/merge/01-merge-changes.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+## Checkstyle
+
+- Jet code was adapted to IMDG checkstyle rules
+- some checkstyle rules were relaxed
+
+There are 2 outstanding items to resolve:
+- some checkstyle rules were ignored for Jet code
+- Jet had some stricker rules regarding public javadoc, this is now not in
+  place, ideally we should bring whole  
+
```
Comment (viliam-durina) [inline docs/design/merge/01-merge-changes.md:65]:
```suggestion
  place, ideally we should apply it to the whole product
```

File: docs/design/merge/01-merge-changes.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+- HazelcastInstance can return JetInstance via #getJetInstance (this will
+  change before the release into a `JetService getJet()` where `JetService`
+  provides a subset of `JetInstance` methods)
+- Jet datastructures created by com.hazelcast.jet.impl.JobRepository are
+  created lazily when needed
+- JetConfig is now a field in `com.hazelcast.config.Config`
+- Jet run tests in parallel by default, Jet tests were marked with `QuickTest`
+  and `ParallelJVMTest` `@Category` accordingly
+- IMDG's `smallInstanceConfig()` for tests sets the
+  `com.hazelcast.jet.config.InstanceConfig#setCooperativeThreadCount` to 2
```
Comment (viliam-durina) [inline docs/design/merge/01-merge-changes.md:54]: Add:
```
- Jet cooperative threads are suspended if no jobs are running.
- One can disable Jet code upload for security reasons.
```

**Review by viliam-durina — CHANGES_REQUESTED** [review]
Missing pieces to describe:
- breaking changes, deprecations
- what happened to CLI (if nothing, mention it)
- what happened to config files

**Review by jbartok — APPROVED** [review]

**Review by viliam-durina — APPROVED** [review]

**Pull request merged**